modules/ipfs-cpinner-client/src/encryption-manager/with-signer.ts

```diff
@@ -0,0 +1,31 @@
+import { IEncryptionManager } from './types'
+import { encrypt, decrypt, generateKeyViaRPC } from './aes'
+import { Web3Provider } from '../web3provider/types'
+
+class EncryptionManager implements IEncryptionManager {
+  private key: string
+  private macKey: string
+
+  constructor (key: string, macKey: string) {
+    this.key = key
+    this.macKey = macKey
+  }
+
+  encrypt = (data: string) => {
+    const cipher = encrypt(this.key, data, this.macKey)
+    return Promise.resolve(Buffer.from(cipher, 'utf-8').toString('hex'))
```

Comment:
cipher should be already an hex string so no need to convert it again

```suggestion
    return Promise.resolve(cipher)
```